cloudsync/tasks.py

```diff
@@ -95,14 +141,42 @@ def transcode_from_s3(self, video_id):  # pylint: disable=unused-argument
 
     # Generate thumbnails for the 1st encoding (no point in doing so for each).
     outputs[0]['ThumbnailPattern'] = THUMBNAIL_PATTERN.format(video_file.s3_basename)
-    transcoder = Transcoder(
+    transcoder = VideoTranscoder(
         settings.ET_PIPELINE_ID,
         settings.AWS_REGION,
         settings.AWS_ACCESS_KEY_ID,
         settings.AWS_SECRET_ACCESS_KEY
     )
-    transcoder.encode(video_input, outputs, Playlists=playlists)
-    transcoder.create_job_for_object(video)
+    try:
+        if not settings.ET_PRESET_IDS:
+            raise ValueError("At least one transcode preset required in settings")
+        transcoder.encode(video_input, outputs, Playlists=playlists)
+    except (ValueError, ClientError,) as exc:
+        log.error('Transcode job creation failed for video %s', video_id)
+        video.update_status(VideoStatus.TRANSCODE_FAILED)
+        self.update_state(task_id=task_id, state=states.FAILURE)
+        if hasattr(exc, 'response'):
+            transcoder.message = exc.response
+        raise
+    finally:
+        transcoder.create_job_for_object(video)
+
+
+@shared_task(bind=True)
+def update_video_statuses(self):  # pylint: disable=unused-argument
+    """
+    Check on statuses of all transcoding videos and update their status if appropriate
+    """
+    transcoding_videos = Video.objects.filter(status=VideoStatus.TRANSCODING)
+    for video in transcoding_videos:
+        try:
+            refresh_status(video)
+        except EncodeJob.DoesNotExist:
+            # Log the exception but don't raise it so other videos can be checked.
+            log.exception("No EncodeJob object exists for video id %d", video.id)
```

Comment:
I don't want to raise an exception here because it might block status checking for other videos.  Sentry.io should alert us of the error in the log, right?